src/visualisation/CSVWriter.cpp

```diff
@@ -4,132 +4,72 @@
+                                           "divergence (1/s)",
+                                           "temperature (Celsius)",
+                                           "concentration (g/m^3)",
+                                           "sight",
+                                           "turb viscosity (m^2/s)",
+                                           "temperature source (K/s)"};
 
-    // Summarize pointers to variables in an array
-    real *vars[] = {static_cast<real *> (u_vel),
-                     static_cast<real *> (v_vel),
-                     static_cast<real *> (w_vel),
-                     static_cast<real *> (pres),
-                     static_cast<real *> (vel_div),
-                     static_cast<real *> (Temp),
-                     static_cast<real *> (Con),
-                     static_cast<real *> (Sight),
-                     static_cast<real *> (turb_visc),
-                     static_cast<real *> (source_T)};
     real* fields[] = {u, v, w, p, div, T, C, s, nu_t, S_T};
     CSVWriter::csv_write(filename, fields, size_vars, var_names);
-
-    // Clean up
-    delete[] (u_vel);
-    delete[] (v_vel);
-    delete[] (w_vel);
-    delete[] (pres);
-    delete[] (vel_div);
-    delete[] (Temp);
-    delete[] (Con);
-    delete[] (Sight);
-    delete[] (turb_visc);
-    delete[] (source_T);
 }
 
-void CSVWriter::csvPrepareAndWrite(const char *filename, real *u, real *v, real *w, real *p, real *T) {
-    Domain *domain = Domain::getInstance();
-    int size = static_cast<int>(domain->get_size());
-
+void CSVWriter::csvPrepareAndWrite(const std::string& filename, real *u, real *v, real *w, real *p, real *T) {
     // Initialize variables
     int size_vars = 5; // Number of variables
-    const char *var_names[] = {"x-velocity (m/s)", "y-velocity (m/s)", "z-velocity (m/s)",
-                               "pressure (kg/(m s^2))",
-                               "temperature (Celsius)"};
-
-    // velocities
-    auto *u_vel = new real[size];
-    auto *v_vel = new real[size];
-    auto *w_vel = new real[size];
-    // pressure
-    auto *pres = new real[size];
-    // temperature
-    auto *Temp = new real[size];
+    std::vector<std::string> var_names = {"x-velocity (m/s)", "y-velocity (m/s)", "z-velocity (m/s)",
+                                           "pressure (kg/(m s^2))",
+                                           "temperature (Celsius)"};
 
     // Summarize pointers to variables in an array
     real* fields[] = {u, v, w, p, T};
 
     CSVWriter::csv_write(filename, fields, size_vars, var_names);
-
-    // Clean up
-    delete[] (u_vel);
-    delete[] (v_vel);
-    delete[] (w_vel);
-    delete[] (pres);
-    delete[] (Temp);
 }
 
-void CSVWriter::csv_write(const char *filename, real **vars, int size_vars, const char **var_names) {
+void CSVWriter::csv_write(const std::string& filename, real **vars, int size_vars,
+        const std::vector<std::string>& var_names) {
+#ifndef BENCHMARKING
```

Comment:
already in time integration. not necessary and in my opinion also not the right place.

Reply:
cannot be resolved without more effort. therefore it won't be changed now